$NetBSD: patch-ae,v 1.1 2008/12/30 15:03:03 he Exp $

Fix for AIX (and possibly others), so that shared libraries will
be found even if there is no corresponding archive library file
present, as is the case with the SAP AG libraries.  Ref.
http://rt.cpan.org/Public/Bug/Display.html?id=41360

--- lib/ExtUtils/Liblist/Kid.pm.orig	2008-10-20 20:18:35.000000000 +0200
+++ lib/ExtUtils/Liblist/Kid.pm	2008-12-01 20:56:29.000000000 +0100
@@ -39,6 +39,7 @@
     my($so)   = $Config{so};
     my($libs) = defined $Config{perllibs} ? $Config{perllibs} : $Config{libs};
     my $Config_libext = $Config{lib_ext} || ".a";
+    my $Config_dlext = $Config{dlext};
 
 
     # compute $extralibs, $bsloadlibs and $ldloadlibs from
@@ -130,8 +131,10 @@
                  && ($Config{'archname'} !~ /RM\d\d\d-svr4/)
 		 && ($thislib .= "_s") ){ # we must explicitly use _s version
 	    } elsif (-f ($fullname="$thispth/lib$thislib$Config_libext")){
+	    } elsif (defined($Config_dlext)
+                 && -f ($fullname="$thispth/lib$thislib.$Config_dlext")){
 	    } elsif (-f ($fullname="$thispth/$thislib$Config_libext")){
-            } elsif (-f ($fullname="$thispth/lib$thislib.dll$Config_libext")){
+	    } elsif (-f ($fullname="$thispth/lib$thislib.dll$Config_libext")){
 	    } elsif (-f ($fullname="$thispth/Slib$thislib$Config_libext")){
 	    } elsif ($^O eq 'dgux'
 		 && -l ($fullname="$thispth/lib$thislib$Config_libext")
